src/dnssec.erl

```diff
@@ -168,7 +168,7 @@ gen_nsec3(RRs, ZoneName, Alg, Salt, Iterations, TTL, Class, Opts) ->
 	 non_neg_integer()) -> binary().
 ih(H, Salt, X, 0) when is_function(H, 1) -> H([X, Salt]);
 ih(H, Salt, X, I) when is_function(H, 1) -> ih(H, Salt, H([X, Salt]), I - 1);
-ih(?DNSSEC_NSEC3_ALG_SHA1, Salt, X, I) -> ih(fun crypto:sha/1, Salt, X, I).
+ih(?DNSSEC_NSEC3_ALG_SHA1, Salt, X, I) -> ih(fun (Data) -> crypto:hash(sha, Data) end, Salt, X, I).
```

Comment:
What is the source of `Data` in the new function definition?

Reply:
Will switch to this form
```suggestion
ih(?DNSSEC_NSEC3_ALG_SHA1, Salt, X, I) -> ih(fun crypto:hash/2, sha, Salt, X, I).
```

Reply:
My concern is that callers only know how to pass a single argument right now. Have you verified that whatever is invoking the function will send two arguments?

Reply:
> Have you verified that whatever is invoking the function will send two arguments?

This effectively curries `crypto:hash/2` into a `/1` function to maintain the interface.

Reply:
Oh, I see. In #11 you create an `ih/5` that accepts the algorithm. My adjustment here curries `crypto:hash/2` into `crypto:hash/1` with the algorithm always `sha`.